src/Symfony/Component/Validator/Constraints/WordCount.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Attribute\HasNamedArguments;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\MissingOptionsException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+final class WordCount extends Constraint
+{
+    public const TOO_SHORT_ERROR = 'cc4925df-b5a6-42dd-87f3-21919f349bf3';
+    public const TOO_LONG_ERROR = 'a951a642-f662-4fad-8761-79250eef74cb';
+
+    protected const ERROR_NAMES = [
+        self::TOO_SHORT_ERROR => 'TOO_SHORT_ERROR',
+        self::TOO_LONG_ERROR => 'TOO_LONG_ERROR',
+    ];
+
+    #[HasNamedArguments]
+    public function __construct(
+        public ?int $min = null,
+        public ?int $max = null,
+        public ?string $locale = null,
+        public string $minMessage = 'This value is too short. It should contain at least one word.|This value is too short. It should contain at least {{ min }} words.',
+        public string $maxMessage = 'This value is too long. It should contain one word or less.|This value is too long. It should contain {{ max }} words or less.',
```

Comment:
Thinking about this again: Is `1` actually a value that makes sense or shouldn't we enforce a value of `2` or higher?

In any case, if we were to keep `1` I propose a small rewording:

```suggestion
        public string $maxMessage = 'This value is too long. It should contain one word.|This value is too long. It should contain {{ max }} words or less.',
```

Reply:
It makes sense to me. I can easily imagine someone having an input where only one word is accepted (like a tag input for instance?). Also "one word" strongly depends on your locale. And it doesn't cost anything to accept this value so I'd say, let's authorize it?

Reply:
works for me 👍